app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ImportExportApplicationService.java

```diff
@@ -0,0 +1,576 @@
+                    Mono.just(defaultPage),
+                    Mono.just(application)
+                );
+            })
+            .flatMap(tuple -> {
+                NewPage defaultPage = tuple.getT1();
+                ApplicationPage defaultPageRef = tuple.getT2();
+                Application application = tuple.getT3();
+
+                defaultPageRef.setId(defaultPage.getUnpublishedPage().getName());
+                final String organizationId = application.getOrganizationId();
+                application.setOrganizationId(null);
+                examplesOrganizationCloner.makePristine(application);
+                String applicationName = application.getName();
+                applicationJson.setExportedApplication(application);
+                return newPageRepository.findByApplicationId(applicationId, AclPermission.READ_PAGES)
+                    .collectList()
+                    .flatMap(newPageList -> {
+                        Map<String, Set<String>> mongoEscapedWidgetsNames = new HashMap<>();
+                        newPageList.forEach(newPage -> {
+                            newPageIdMap.put(newPage.getId(), newPage.getUnpublishedPage().getName());
+                            newPage.setApplicationId(applicationName);
+                            examplesOrganizationCloner.makePristine(newPage);
+
+                            if (newPage.getUnpublishedPage().getLayouts() != null) {
+                                newPage.getUnpublishedPage().getLayouts().forEach(layout ->
+                                    mongoEscapedWidgetsNames.put(layout.getId(), layout.getMongoEscapedWidgetNames())
+                                );
+                            }
+
+                            if (newPage.getPublishedPage() != null
+                                    && newPage.getPublishedPage().getLayouts() != null) {
+
+                                newPage.getPublishedPage().getLayouts().forEach(layout ->
+                                    mongoEscapedWidgetsNames.put(layout.getId(), layout.getMongoEscapedWidgetNames())
+                                );
+                            }
+                        });
+                        applicationJson.setPageList(newPageList);
+                        applicationJson.setMongoEscapedWidgets(mongoEscapedWidgetsNames);
+                        return datasourceRepository
+                            .findAllByOrganizationId(organizationId, AclPermission.READ_DATASOURCES)
+                            .collectList();
+                    })
+                    .map(datasourceList -> {
+                        datasourceList.forEach(datasource -> datasourceMap.put(datasource.getId(), datasource.getName()));
+                        applicationJson.setDatasourceList(datasourceList);
+                        return newActionRepository.findByApplicationId(applicationId);
+                    })
+                    .flatMap(Flux::collectList)
```

Comment:
I think there's double-wrapping in the return type of `.map` here. As in, it returns a `Mono<Flux<List>>`. This should avoid that.

```suggestion
                    .flatMapMany(datasourceList -> {
                        datasourceList.forEach(datasource -> datasourceMap.put(datasource.getId(), datasource.getName()));
                        applicationJson.setDatasourceList(datasourceList);
                        return newActionRepository.findByApplicationId(applicationId);
                    })
                    .collectList()
```

Reply:
Updated code pushed!